src/pal/src/misc/stackstring.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+
+#include "pal/malloc.hpp"
+
+template <UINT32 MEMSIZE>
+class StackString
+{
+private:
+	BYTE m_innerBuffer[MEMSIZE];
+	BYTE *m_buffer;
+	UINT32 m_allocation; // actual allocated size
+	BOOL m_isAllocated;
+
+public:
+	StackString(UINT32 size)
+	: m_allocation(0), m_buffer(NULL), m_isAllocated(false)
+	{
+		ReallocateBuffer(size);
+	}
+
+	void Set(const BYTE *buffer, UINT32 size)
```

Comment:
You should also add a private copy constructor (or use some other scheme) to avoid accidently using the copy constructor autogenerated by the compiler.